components/normalizer/src/lib.rs

```diff
@@ -1630,16 +1729,63 @@ impl DecomposingNormalizer {
         })
     }
 
-    icu_provider::gen_any_buffer_constructors!(
-        locale: skip,
-        options: skip,
-        error: NormalizerError,
-        functions: [
-            Self::try_new_nfkd_unstable,
-            try_new_nfkd_with_any_provider,
-            try_new_nfkd_with_buffer_provider
-        ]
-    );
+    #[doc(hidden)]
+    #[cfg(all(feature = "experimental", feature = "data"))]
+    pub const fn new_uts46_decomposed_without_ignored_and_disallowed() -> Self {
```

Comment:
Nit: You should be able to auto-generate this, right? It's just calling `_unstable` with `crate::provider::Baked`? I guess the difference is that you avoid the singleton check; is that it?

Reply:
Auto generated constructors will go through `DataProvider::load` and be fallible. This uses singletons and is infallible.